packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -421,3 +422,70 @@ export async function saveAllIssuers(zcf, issuerKeywordRecord = {}) {
   );
   return Promise.all(issuersPSaved);
 }
+
+/** @type {MapKeywords} */
+export const mapKeywords = (keywordRecord = {}, keywordMapping) => {
+  return Object.fromEntries(
+    Object.entries(keywordRecord).map(([keyword, value]) => {
+      if (keywordMapping[keyword] === undefined) {
+        return [keyword, value];
+      }
+      return [keywordMapping[keyword], value];
+    }),
+  );
+};
+/** @type {Reverse} */
+const reverse = (keywordRecord = {}) => {
+  return Object.fromEntries(
+    Object.entries(keywordRecord).map(([key, value]) => [value, key]),
+  );
+};
+
+/** @type {OfferTo} */
+export const offerTo = async (
+  zcf,
+  invitation,
+  keywordMapping = {},
+  proposal,
+  fromSeat,
+  toSeat,
+) => {
+  const zoe = zcf.getZoeService();
+  const mappingReversed = reverse(keywordMapping);
+
+  // the proposal is in the other contract's keywords, but we want to
+  // use `proposal.give` to withdraw
+  const payments = await withdrawFromSeat(
+    zcf,
+    fromSeat,
+    // `proposal.give` may be undefined
+    mapKeywords(proposal.give, mappingReversed),
+  );
+
+  // Map to the other contract's keywords
+  const paymentsForOtherContract = mapKeywords(payments, keywordMapping);
+
+  const userSeatPromise = E(zoe).offer(
+    invitation,
+    proposal,
+    paymentsForOtherContract,
+  );
+
+  const depositedPromiseKit = makePromiseKit();
+
+  const doDeposit = async payoutPayments => {
+    const amounts = await E(userSeatPromise).getCurrentAllocation();
```

Comment:
I thought this wouldn't work if the contract exited, but it looks like the Zoe seats survive after the contract vat shuts down. That's kinda nice.  I hadn't realized it.

Reply:
Yeah, the ZoeSeatAdmin is removed, but not the userSeat. The user might be getting a payout still.